src/api_client.js

```diff
@@ -173,6 +188,27 @@ export class ApiClient {
       });
   }
 
+  fetchJWTToken = async () => {
+    let result = { token: null, errors: null };
+
+    try {
+      const response = await fetch('/apps/submarine/auth/tokens');
```

Comment:
I think this is good for now. I can see there being a need to make this endpoint configurable based on the environment (as we may have a proxy at `/apps/submarine-staging` for example), but until it's a problem this is cool.